x/emissions/keeper/msg_server.go

```diff
@@ -78,10 +86,20 @@ func (ms msgServer) CreateNewTopic(ctx context.Context, msg *state.MsgCreateNewT
 }
 
 func (ms msgServer) SetWeights(ctx context.Context, msg *state.MsgSetWeights) (*state.MsgSetWeightsResponse, error) {
```

Comment:
unit test should now add a failing case to ensure that unauthorized senders are disallowed